### ISSUE TITLE
请问一下层次hard mining体现在代码的哪里？

### ISSUE THREAD

**Issue by iridescent303** [YiYuanIntelligent/3DFasterRCNN_LungNoduleDetector#1]
之前看技术圈的解释，没看懂。

**Comment by YiYuanIntelligent** [comment]
hardmining 的代码再这块。https://github.com/YiYuanIntelligent/3DFasterRCNN_LungNoduleDetector/blob/master/lib/rpn/tianchi_anchor_target_layer.py#L97

**Comment by YiYuanIntelligent** [comment]
hardmining 的代码在这块。https://github.com/YiYuanIntelligent/3DFasterRCNN_LungNoduleDetector/blob/master/lib/rpn/tianchi_anchor_target_layer.py#L97  。层次化hard mining是工程上的一种实现，没有放在网络里面，我们会先训练一个模型，作为第一层模型。然后以这个模型的基础上，finetune比上一层模型更难分的数据，然后作为第二层模型，依次往下。

**Comment by iridescent303** [comment]
哦，是把第一次得到的bbox放到第二次进行训练是吧

**Comment by iridescent303** [comment]
对了，你们这里的batchnorm层后面为什么不加scale层啊，感觉这样和pytorch里面的不一样了。

**Comment by YiYuanIntelligent** [comment]
我记得当时，逐层核对过caffe和pytorch的输出，batchnorm这块的实现，两者有挺大的差异。我们应该是尝试过，加或不加，caffe都无法正常收敛，所以后面没管这层了

**Comment by iridescent303** [comment]
你们固定了输入比较了输出吗，我实现的加了scale输出是一样的。

**Comment by YiYuanIntelligent** [comment]
150层的结果都一样吗？bn那块的参数是怎么设置的

**Comment by iridescent303** [comment]
都用的默认的参数，不过我是保持全部权重一致。

**Comment by iridescent303** [comment]
不过bn层参数caffe和pytorch的不太一样

**Comment by zp678** [comment]
Hello, I want to know how to make a data set about this network

**Comment by zp678** [comment]
Hello, I want to know how to create a data set for this network. I didn't participate in the Tianchi competition so I can't download it. Can you share it?
